loader/loader.c

```diff
@@ -5376,6 +5420,17 @@ VKAPI_ATTR VkResult VKAPI_CALL terminator_CreateInstance(const VkInstanceCreateI
     bool one_icd_successful = false;
 
     struct loader_instance *ptr_instance = (struct loader_instance *)*pInstance;
+    if (NULL == ptr_instance) {
+        loader_log(ptr_instance, VULKAN_LOADER_ERROR_BIT, 0,
+                   "terminator_CreateInstance: Instance pointer (NULL) passed down appears invalid. Possibly corrupted by active "
+                   "layer (Policy #LLP_LAYER_21)");
+    } else if (LOADER_MAGIC_NUMBER != ptr_instance->magic) {
+        loader_log(ptr_instance, VULKAN_LOADER_ERROR_BIT, 0,
+                   "terminator_CreateInstance: Instance pointer (%p) passed down appears invalid with MAGIC value 0x%08x. Possibly "
+                   "corrupted by active layer (Policy #LLP_LAYER_21).  ",
```

Comment:
"Appears invalid with MAGIC value" Not very clear what this means. I read 'appears invalid with magic value' as it saying its invalid because it has a magic value, not because it lacks it.